File: solutions/python/twelve-days/1/twelve_days.py
days = {
    1: "first",
    2: "second",
    3: "third",
    4: "fourth",
    5: "fifth",
    6: "sixth",
    7: "seventh",
    8: "eighth",
    9: "ninth",
    10: "tenth",
    11: "eleventh",
    12: "twelfth",
}

gifts = {
    1: "a Partridge in a Pear Tree.",
    2: "two Turtle Doves,",
    3: "three French Hens,",
    4: "four Calling Birds,",
    5: "five Gold Rings,",
    6: "six Geese-a-Laying,",
    7: "seven Swans-a-Swimming,",
    8: "eight Maids-a-Milking,",
    9: "nine Ladies Dancing,",
    10: "ten Lords-a-Leaping,",
    11: "eleven Pipers Piping,",
    12: "twelve Drummers Drumming,",
}


def recite(start_verse, end_verse):
    verses = []

    for day in range(start_verse, end_verse + 1):
        line = f"On the {days[day]} day of Christmas my true love gave to me: "

        parts = []
        for gift_day in range(day, 0, -1):
            if gift_day == 1 and day > 1:
                parts.append("and " + gifts[gift_day])
            else:
                parts.append(gifts[gift_day])

        line += " ".join(parts)
        verses.append(line)

    return verses

File: solutions/javascript/twelve-days/1/twelve-days.js
//
// This is only a SKELETON file for the 'Twelve Days' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

let gifts = [
  'a Partridge in a Pear Tree.',
  'two Turtle Doves',
  'three French Hens',
  'four Calling Birds',
  'five Gold Rings',
  'six Geese-a-Laying',
  'seven Swans-a-Swimming',
  'eight Maids-a-Milking',
  'nine Ladies Dancing',
  'ten Lords-a-Leaping',
  'eleven Pipers Piping',
  'twelve Drummers Drumming',
];

let days = [
  'first',
  'second',
  'third',
  'fourth',
  'fifth',
  'sixth',
  'seventh',
  'eighth',
  'ninth',
  'tenth',
  'eleventh',
  'twelfth',
];

export function recite(startVerse, endVerse = startVerse) {
  let result = [];

  for (let i = startVerse; i <= endVerse; i++) {
    let line = `On the ${days[i - 1]} day of Christmas my true love gave to me: `;

    let dailyGifts = [];
    for (let j = i - 1; j >= 0; j--) {
      if (j == 0 && i > 1) {
        dailyGifts.push('and ' + gifts[0]);  // fixed missing space + variable typo
      } else {
        dailyGifts.push(gifts[j]);
      }
    }

    line += dailyGifts.join(', ') + '\n';
    result.push(line);
  }

  if (endVerse === startVerse) {
    return result[0];
  } else {
    return result.join('\n');
  }
}


         

